temporal-sdk/src/main/java/io/temporal/common/converter/SearchAttributesPayloadConverter.java

```diff
@@ -0,0 +1,52 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.common.converter;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.protobuf.ByteString;
+import io.temporal.api.common.v1.Payload;
+import io.temporal.internal.common.SearchAttributesUtil;
+import java.util.Optional;
+
+public class SearchAttributesPayloadConverter extends JacksonJsonPayloadConverter {
+
+  public SearchAttributesPayloadConverter() {
+    super();
+  }
+
+  @Override
+  public Optional<Payload> toData(Object value) throws DataConverterException {
```

Comment:
Please rework this class from inheritance to composition. In general, always prefer composition over inheritance. The fact that this method doesn't delegate to super is one more argument for composition here.
Also consider not using JacksonJsonPayloadConverter at all. This method is fully redefined. Mapper should be defined here too. And fromData should be redefined too. BTW, why is it not?
We don't need anything from the parent class at all.